Nfultz/refactor hof2 cite summary

nfultz opened the pull request:
The main goal of this refactor was to funnel as much NSE through a single place, `declaration_template` - some collateral changes are:

* All delegate functions  now called `delegate`
* All steps can be labeled, labels should be text (**named steps)
* citation functionality is now a step, `declare_citation`
* `design` data structure internals rewritten, list of objects rather than parallel arrays
  * `design$design_function()` gone, use `execute_design(design)`
  * modify_design wrapper not needed, instead use `insert/delete/replace_step` directly.

nfultz committed:
Basic designs somewhat back to working.

nfultz committed:
population and estimands working.

nfultz committed:
subset now working.

nfultz committed:
Fixing custom functions.

nfultz committed:
Fixing test-factorial.R

nfultz committed:
Fixing test-diagnosands.

nfultz committed:
fix test-bootrap-diagnosands.

nfultz committed:
fix test-declare-design

nfultz committed:
Fix test-environment-issues

nfultz committed:
fixing demo and download.

nfultz committed:
Fixing more tests.

nfultz committed:
Fixing estimator

nfultz committed:
removing modify_design from the estimator test.

nfultz committed:
cite is now a step; summary working again

nfultz committed:
Updating modification helper functions.

nfultz commented:
Future work 
- streamline (PO, Assign, Reveal) triples of steps st reveal is no longer explicitly required in the default case.
- updating vignettes
- purging dead code that is currently commented out

graemeblair commented:
This is great! I'm not sure we need a user-exposed `execute_design` function, since you can get those from the more useful `get_estimates` and `get_estimands` if I understand correctly. 

Still thinking on better name for `delegate`!

nfultz committed:
Fixing vignette for build

nfultz committed:
Update custom_functions.Rmd

nfultz committed:
s/delegate/handler

nfultz committed:
fixing for _var change in randomizer

nfultz committed:
Merge branch 'nfultz/refactor_hof2_cite_summary' of https://github.com/DeclareDesign/DeclareDesign into nfultz/refactor_hof2_cite_summary

nfultz committed:
s/delegate/handler/

nfultz committed:
s/level/add_level/

nfultz committed:
Merge branch 'master' into nfultz/fix_vignette

nfultz committed:
Fix handler

nfultz committed:
Merge branch 'nfultz/refactor_hof2_cite_summary' into nfultz/fix_vignette

nfultz committed:
fixing build; fabricatr now wants add_level, handler, etc

nfultz committed:
Update custom_functions.Rmd (#76)

* Documentation pass Nov 29th 2017

* Documentation update Nov 30 2017

* Removed conditional output in vignette.

* bug fix

* Requested documentation change Dec 6 2017

* remove sf for now given travis issues (#66)

* remove sf for now given travis issues
* Removes the installation of gdal dependencies.
* Update two_arm.Rmd
* Update DESCRIPTION (#68)
* Trying to force appveyor to use drat - when it chooses to install from github, estimatr fails bc rtools is not installed.
* Instead of $PKG_TARBALL, which is the output of the internal R CMD build on travis, use the packages that are explicitly built by the after_success step.
* Move autogenerated zip binary up a folder, so that deploy.sh can find it :sigh:
* Install drat where needed.
* Stealing logic from https://github.com/openmeeg/openmeeg/blob/master/.appveyor.yml#L63 to only build on pushes to master.
* Instead of $PKG_TARBALL, which is the output of the internal R CMD build on travis, use the packages that are explicitly built by the after_success step.
* Move autogenerated zip binary up a folder, so that deploy.sh can find it :sigh:
* Install drat where needed - and use mavericks/el-capitan path on mac as appropriate
* new token for travis also
* downgrading osx oldrel to closest thing to mavericks
* point linux oldrel at nloptr

* Update .travis.yml

drat was not found?

* Update .travis.yml

factor out pkgdown

* Update .travis.yml

Don't package and deploy source build from oldrel

* Nfultz/ci ddtools (#71)

replace deploy.sh with DDtools::update_drat now that drat v1.4 is compatible with R 3.4 and master branch.

* paper up

* Update DESCRIPTION

* Update DESCRIPTION

* Delete deploy.sh (#72)

* Update custom_functions.Rmd

* Delete paper.pdf

* paper up

* s/delegate/handler/

* s/level/add_level/

* Fix handler

* fixing build; fabricatr now wants add_level, handler, etc

nfultz committed:
Merge branch 'master' into nfultz/refactor_hof2_cite_summary

nfultz committed:
update to new version of helper function from fabricatr

nfultz committed:
Fixing PO test, downgrading to 0.2.0

nfultz committed:
Merge branch 'nfultz/refactor_hof2_cite_summary' of https://github.com/DeclareDesign/DeclareDesign into nfultz/refactor_hof2_cite_summary

nfultz commented:
Finally got a green check, merging.
